### PR TITLE
Support thread messages as attachments

### PR DESCRIPTION
**Pull request by renatobecker**
Before the implementation of threads, the payload of `reply` and `quoted` messages were equal, but now the reply messages are not sent as attachments anymore, so we need to fetch the parent message from the server and then render it as an attachment.

### CODE REVIEW

File: src/lib/room.js [hunk 1]
```diff
@@ -117,7 +122,7 @@ export const loadMessages = async() => {
 	}
 
 	await store.setState({ loading: true });
-	const messages = await Livechat.loadMessages(rid);
+	const messages = await Livechat.loadMessages(rid).then(async(data) => (normalizeMessages(data)));
```
Comment (diegolmello) [inline src/lib/room.js]: Unnecessary async

File: src/lib/room.js [hunk 2]
```diff
@@ -136,7 +141,7 @@ export const loadMoreMessages = async() => {
 	}
 
 	await store.setState({ loading: true });
-	const moreMessages = await Livechat.loadMessages(rid, { limit: messages.length + 10 });
+	const moreMessages = await Livechat.loadMessages(rid, { limit: messages.length + 10 }).then(async(data) => (normalizeMessages(data)));
```
Comment (diegolmello) [inline src/lib/room.js]: Unnecessary async

File: src/lib/threads.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	let parentMessage = messages.find((msg) => msg._id === message.tmid);
+	if (!parentMessage) {
+		parentMessage = await findParentMessage(message.tmid);
+	}
+	const { msg, attachments = [] } = parentMessage;
+	return Object.assign(message, { threadMsg: parentMessage, attachments: [{ attachments, text: msg, tmid: message.tmid }] });
+};
+
+export const normalizeMessage = async(message) => {
+	if (await isThreadMessage(message)) {
```
Comment (diegolmello) [inline src/lib/threads.js]: Feels weird to me.
Can you run `await isThreadMessage(message)` first and then test the result?

File: src/lib/threads.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		return false;
+	}
+
+	if (message && message.tmid && !message.threadMsg) {
+		return await normalizeThreadMessage(message);
```
Comment (diegolmello) [inline src/lib/threads.js]: Unnecessary await

File: src/lib/threads.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	return message;
+};
+
+export const normalizeMessages = async(messages = []) => await Promise.all(messages.filter(async(message) => normalizeMessage(message)));
```
Comment (diegolmello) [inline src/lib/threads.js]: The second async is unnecessary

**Comment by renatobecker** [comment]
@diegolmello, fixed!
Thanks.

**Pull request merged**